src/allmydata/storage_client.py

```diff
@@ -1170,6 +1247,9 @@ def remove_connecting_deferred(result):
             self._failed_to_connect
         )
 
+        # TODO Make sure LoopingCall waits for the above timeout for looping again:
```

Comment:
Do we need to resolve this TODO here?  But ... this was the function used with a LoopingCall _before_ this change too and had no return value... so it's not worse.  And maybe it is even a bit better now because the earlier async work will be captured by the Deferred created by `async_to_deferred`.

But ... why not await this Deferred too?

Reply:
This is a thing that would be nice to do in the future but was pre-existing issue. Fixing it causes tests to fail with dirty reactor and I didn't want to get into that in this PR.

Reply:
I filed https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3981